src/wp-admin/includes/dashboard.php

```diff
@@ -1555,3 +1558,251 @@ function wp_welcome_panel() {
+ *
+ * @since 1.0.0
+ */
+function cp_dashboard_petitions_table_body ( $votesCount, $link, $title, $author, $status, $createdTime ) {
+?>
+	<tr>
+		<td class="votes-count"><?php echo esc_attr( $votesCount ); ?></td>
+
+		<td class="petition">
+			<a target="_blank" href="<?php echo esc_url( $link ) ?>"><strong><?php echo esc_attr__( $title )?><span class="screen-reader-text"><?php echo esc_attr__( '(opens in a new window)' ); ?></span><span aria-hidden="true" class="dashicons dashicons-external"></span></strong></a>
+			<?php
+				esc_attr__( 'by' ) . ' ' . ucwords(  esc_attr( $author ) );
+
+				if ( $status == "open" ){
+					echo esc_attr__( ' - ' ) . ' ' . human_time_diff( strtotime( $createdTime ), current_time('timestamp') ) . ' ' . esc_attr__( 'ago' );
+				} 
+				elseif ( $status == "planned" ){
+					echo ' - ' . '<span class="planned">' . esc_attr__( ucfirst( $status ) ) . '</span>';
+				} 
+				else{
+					echo ' - ' . '<span class="started">' . esc_attr__( ucfirst( $status ) ) . '</span>';
+				}
+			?>
+		</td>
+	</tr>
+<?php
+}
+	
+/**
+ * Callback function for the petitions dashboard widget
+ *
+ * @since 1.0.0
+ */
+function cp_dashboard_petitions() {
+
+	// Get any existing copy of our transient data
+	if ( false === ( $dashboard_petitions_results = get_transient( 'dashboard_petitions_results' ) ) ) {
+		/**
+		 * It wasn't there, so regenerate the data and save the transient
+		 * Query API for JSON data -> decode results to php
+		 *
+		 * @since 1.0.0
+		 * @return array
+		 */
+		$api_url = 'https://api-v1.classicpress.net/features/1.0/';
+
+		/**
+		 * Response should be an array with:
+		 *  'most-wanted []' - array() - A user-friendly platform name, if it can be determined
+		 *  'trending []' - array() - A user-friendly browser name
+		 *  'recent []' - array() - The version of the browser the user is using
+		 *  'tags [] - array() - The version of the browser the user is using
+		 *  'link - string - The version of the browser the user is using
+		 */
+		$response = wp_remote_get( $api_url );
+
+		if ( ! is_array( $response ) ) {
+			return;
+		}
+
+		if ( is_array( $response ) ) {
+			$dashboard_petitions_results = $response['body']; // use the content
+		}
+
+		set_transient( 'dashboard_petitions_results', $dashboard_petitions_results, 12 * HOUR_IN_SECONDS );
+	}
+	
+	/** 
+	 * Parse URL Ouput into PHP array
+	 * 
+	 * @since 1.0.0
+	 * @return array
+	 */
+	$json = json_decode( $dashboard_petitions_results, true );
+	
+	/**
+	 * Response should be an array with:
+	 * 'votes' - Count Up Voted of petition
+	 * 'link' - URL to the petitions website page 
+	 * 'title' - Title of Petition 
+	 * 'author' - Name of Petition Lead 
+	 * 'status' - Status of petition  
+	 * 'createdTime' - Timestamp of the petition request 
+	 */
+	$most_wanted = $json['most-wanted']; // get most wanted (upvoted)
+	$recent = $json['recent']; // get recent
+	$trending = $json['trending']; // get trending
+
+	/** 
```

Comment:
@bahiirwa Would be good to configure your editor to automatically remove trailing spaces.